core/src/nucleus/actions/validate.rs

```diff
@@ -43,7 +43,17 @@ pub fn validate_entry<'a>(
             .nucleus()
             .dna()
             .unwrap()
-            .get_zome_name_for_entry_type(&entry.entry_type().to_string())
+            .get_zome_name_for_app_entry_type(match &entry.entry_type() {
+                EntryType::App(app_entry_type) => app_entry_type,
+                _ => {
+                    return FutureObj::new(Box::new(future::err(HolochainError::ValidationFailed(
+                        format!(
+                            "Attempted to validate system entry type {:?}",
+                            entry.entry_type()
+                        ),
+                    ))));
+                }
+            })
```

Comment:
Hm, these lines would be the reason imo to change `fn validate_entry` to be `async` which would mean we could replace this `return FutureObj...` with a `?`..

Reply:
Would also be nice to have a function:
```rust
fn to_app_entry_type() -> Result<String, HolochainError>
```
to change this to
```
.get_zome_name_for_entry_type(&entry.entry_type().to_app_entry_type()?)
```